src/include/columnar/columnar_tableam.h

```diff
@@ -8,6 +8,44 @@
 
 #include "distributed/coordinator_protocol.h"
 
+
+/*
+ * Number of valid ItemPointer Offset's for "row number" <> "ItemPointer"
+ * mapping.
+ *
+ * Postgres has some asserts calling either ItemPointerIsValid or
+ * OffsetNumberIsValid. That constraints itemPointer.offsetNumber
+ * for columnar tables to the following interval:
+ * [FirstOffsetNumber, MaxOffsetNumber].
+ *
+ * However, for GIN indexes, Postgres also asserts the following in
+ * itemptr_to_uint64 function:
+ * "GinItemPointerGetOffsetNumber(iptr) < (1 << MaxHeapTuplesPerPageBits)",
+ * where MaxHeapTuplesPerPageBits = 11.
+ * That means, offsetNumber for columnar tables can't be equal to
+ * 2**11 = 2048 = MaxOffsetNumber.
+ * Hence we can't use MaxOffsetNumber as offsetNumber too.
```

Comment:
FYI:
What does that mean ? Below was throwing an assertion failure on index support branch. Then, as written above, I decided not to use `MaxOffsetNumber` as offset number:

```sql
CREATE TABLE testjsonb (
       j jsonb
) using columnar;
INSERT INTO testjsonb SELECT CAST('{"f1" : ' ||'"'|| i*4 ||'", ' || '"f2" : '||'"'|| i*10 ||'"}' as json) FROM generate_series(1,1012) i;
INSERT INTO testjsonb SELECT CAST('{"f1" : ' ||'"'|| i*4 ||'", ' || '"f2" : '||'"'|| i*10 ||'"}' as json) FROM generate_series(1,1012) i;
INSERT INTO testjsonb SELECT CAST('{"f1" : ' ||'"'|| i*4 ||'", ' || '"f2" : '||'"'|| i*10 ||'"}' as json) FROM generate_series(1,1012) i;
INSERT INTO testjsonb SELECT CAST('{"f1" : ' ||'"'|| i*4 ||'", ' || '"f2" : '||'"'|| i*10 ||'"}' as json) FROM generate_series(1,1012) i;
CREATE INDEX jidx ON testjsonb USING gin (j);
```

Reply:
I posted to pgsql-hackers to get some clarity here. Good find!